migrator/migrations/m_170_to_m_171_move_scope_to_collection_in_report_configurations/migration.go

```diff
@@ -0,0 +1,247 @@
+	accessScopePostgres "github.com/stackrox/rox/migrator/migrations/m_170_to_m_171_move_scope_to_collection_in_report_configurations/accessScopePostgresStore"
+	collectionPostgres "github.com/stackrox/rox/migrator/migrations/m_170_to_m_171_move_scope_to_collection_in_report_configurations/collectionPostgresStore"
+	reportConfigurationPostgres "github.com/stackrox/rox/migrator/migrations/m_170_to_m_171_move_scope_to_collection_in_report_configurations/reportConfigurationPostgresStore"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+const (
+	batchSize = 500
+
+	startSeqNum = 170
+
+	skippedMigrationMessageTemplate = "Failed to migrate report configuration '%s': Attached scope <%s> has label selector/s " +
+		"with a different operator than 'IN'; Resolution: %s"
+
+	manualInterventionInstruction = "Please manually create a collection and add it to the report configuration. " +
+		"The report will stop working until a collection is added to it."
+
+	embeddedCollectionTemplate = "System-generated embedded collection %d for scope <%s>"
+	rootCollectionTemplate     = "System-generated root collection for scope <%s>"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	migration = types.Migration{
+		StartingSeqNum: startSeqNum,
+		VersionAfter:   &storage.Version{SeqNum: int32(startSeqNum + 1)}, // 171
+		Run: func(databases *types.Databases) error {
+			err := moveScopeIDToCollectionIDInReports(databases.PostgresDB)
+			if err != nil {
+				return errors.Wrap(err, "error converting scopes to collections in reportConfigurations")
+			}
+			return nil
+		},
+	}
+
+	scopeIDToCollectionID = make(map[string]string)
+	skippedScopes         = set.NewStringSet()
+	idGenerator           func(collectionName string) string
+)
+
+func buildEmbeddedCollection(scopeName string, index int, rules []*storage.SelectorRule) *storage.ResourceCollection {
+	timeNow := protoconv.ConvertTimeToTimestamp(time.Now())
+	colName := fmt.Sprintf(embeddedCollectionTemplate, index, scopeName)
+	return &storage.ResourceCollection{
+		Id:          idGenerator(colName),
+		Name:        colName,
+		CreatedAt:   timeNow,
+		LastUpdated: timeNow,
+		ResourceSelectors: []*storage.ResourceSelector{
+			{
+				Rules: rules,
+			},
+		},
+	}
+}
+
+func labelSelectorsToCollections(scopeName string, index int, labelSelectors []*storage.SetBasedLabelSelector, fieldName string) ([]*storage.ResourceCollection, bool) {
+	collections := make([]*storage.ResourceCollection, 0, len(labelSelectors))
+	for _, labelSelector := range labelSelectors {
+		selectorRules := make([]*storage.SelectorRule, 0, len(labelSelector.GetRequirements()))
+		for _, requirement := range labelSelector.GetRequirements() {
+			if requirement.GetOp() != storage.SetBasedLabelSelector_IN {
+				return nil, false
```

Comment:
returning bool hides the root cause in the caller. Instead return error, for example, in this case `unsupported...` err msg